package.json

```diff
@@ -36,39 +36,51 @@
 		"clean": "lerna run clean",
 		"ganache": "npx ganache -m 'smart guide what forget tired jungle always expire rescue boring glue champion' --account=\"0x4c3758228f536f7a210f8936182fb5b728046970b8e3215d0b5cb4c4faae8a4e,100000000000000000000\" --account=\"0x34aeb1f338c17e6b440c189655c89fcef148893a24a7f15c0cb666d9cf5eacb3,100000000000000000000\"",
 		"ganache-background": "yarn ganache > /dev/null & client=$! npx wait-port 8545",
+		"geth:install-ubuntu": "sudo add-apt-repository -y ppa:ethereum/ethereum && sudo apt-get update && sudo apt-get install ethereum && geth init --datadir .github/geth/data `pwd`/.github/geth/genesis.json && sudo apt install screen",
 		"geth:build": "docker build ./.github -t gethnode -f .github/Dockerfile",
-		"geth": "docker run -d -p 8545:8545 gethnode --datadir data --networkid 15 --http.addr 0.0.0.0 --http.port 8545 --http",
+		"geth-local": "screen -DmS geth geth --datadir .github/geth/data --networkid 15 --http.addr 0.0.0.0 --http.port 8545 --http --ws --ws.addr 0.0.0.0 --ws.port 8545 --ws.origins '*' --ipcpath=./.github/geth/data/geth.ipc </dev/null &>/dev/null &",
```

Comment:
WS port should not be different than http port?

Reply:
both ganache and geth can run ws and HTTP on the same port, just for simplicity. Should I undo the change?
@avkos

Reply:
yes it works on the same port because it is different protocol